langserver/references.go

```diff
@@ -181,10 +185,51 @@ func (h *LangHandler) handleTextDocumentReferences(ctx context.Context, conn JSO
 		lconf.Load() // ignore error
 	}()
 
-	// Wait for timeout or completion
-	select {
-	case <-done:
-	case <-ctx.Done():
+	streamPos := 0
+	streamTick := make(<-chan time.Time, 1)
+	if streamExperiment {
+		t := time.NewTicker(time.Second)
+		defer t.Stop()
+		streamTick = t.C
+	}
+Loop:
+	for {
+		select {
+		case <-done:
+			break Loop
+		case <-ctx.Done():
+			break Loop
+		case <-streamTick:
+		}
+
+		// Send partial results
+		mu.Lock()
+		partial := refs
+		mu.Unlock()
+		if len(partial) == streamPos {
+			continue
+		}
+
+		initial := streamPos == 0
+		patches := make([]lspext.JSONPatch, len(partial)-streamPos, 0)
+		for ; streamPos < len(partial); streamPos++ {
+			patches = append(patches, lspext.JSONPatch{
+				OP:    "add",
+				Path:  "/-",
+				Value: goRangeToLSPLocation(fset, partial[streamPos].Pos(), partial[streamPos].End()),
+			})
+		}
+		if initial {
+			patches[0].Path = ""
```

Comment:
This only sets the path, what about the value?
The first operation must be `{"op": "add", "path": "", "value": []}`, to initialize the result to an empty array on the client. Otherwise the client will not be able to apply further pushes the array, because the result will not be an array, but `undefined`.